core/src/main/scala/fs2/job/JobManager.scala

```diff
@@ -68,7 +68,17 @@ final class JobManager[F[_]: Concurrent: Timer, I, N] private (
     val putStatusF = Concurrent[F] delay {
       val attempt = Context[F](Status.Pending, None)
 
-      Option(meta.putIfAbsent(job.id, attempt)).isEmpty
+      val canceledToPending =
```

Comment:
This is a good idea, but due to how job submission works, unfortunately I don't think it achieves the desired outcome.

The reason the current implementation treats `Canceled` as a terminal state is because a job that was canceled while pending still exists in the job queue. `Canceled` acts as a sentinel value so that when the canceled job is eventually dequeued and run, it will notice it is canceled and immediately terminate.

Transitioning from `Canceled` back to `Pending` will result in the following scenario in the case where the job manager has capacity to run both jobs in parallel

1. job 1a is submitted, enqueued and has pending status
2. prior to job 1a being run, it is canceled and now has canceled status, but is still in the job queue
3. job 1b is submitted, transitioning the state of job '1' from canceled to pending and enqueued
4. the job manager has free capacity and dequeues the next job
5. job 1a is dequeued, checks its status and sees it is pending, so it starts
6. job 1b is dequeued, checks its status and sees it is running, so it errors that the job already exists

In the degenerate case of a completely serial job manager (like the tests), (5) would happen and then when the job finished it would remove the id from the manager. When 1b is dequeued in (6) it would notice there is no such job in the map and abort execution.

In either case, the result will be that the job that was intended to be canceled will be run and the subsequent job will not run.

Reply:
Oh, right - I totally missed that! Thank you for pointing that out.